File: my-folder/0020-valid-parentheses/solution.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> parent = new Stack<Character>();
		for (int i = 0; i < s.length(); i++) {
			if (s.charAt(i) == '(' || s.charAt(i) == '[' || s.charAt(i) == '{' || s.charAt(i) == '<') {
				parent.push(s.charAt(i));
			} else {
				if (parent.empty()) {
					return false;
				} else if ((parent.peek() == '(' && s.charAt(i) != ')') || (parent.peek() == '[' && s.charAt(i) != ']')
						|| (parent.peek() == '{' && s.charAt(i) != '}')
						|| (parent.peek() == '<' && s.charAt(i) != '>')) {
					return false;
				} else {
					parent.pop();
				}
			}
		}
		if (!parent.empty()) {
			return false;
		}
		return true;
    }
}

File: my-folder/0020-valid-parentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for(int i=0;i<s.length();i++)
        {
            char temp = s[i]; 
            if(temp=='(' || temp=='{' || temp=='[')st.push(temp);
            else
            {
                if(st.empty())return false;
                if((temp==')' && st.top()!='(') || (temp=='}' && 
                st.top()!='{') || (temp==']' && st.top()!='['))return false;
                st.pop();
            }
        }
        return st.empty();
    }
};
